File: prototype/python/shape.py
from abc import ABC, abstractmethod

class Shape(ABC):
  x = None
  y = None
  color = None
  
  def __init__(self, params = None):
    if params is not None:
      self.x = params['x']
      self.y = params['y']
      self.color = params['color']

  @abstractmethod
  def clone():
    pass

File: Generating/prototype/src/Shape.ts
import { ShapeParams, IShape } from './interfaces'

export default abstract class Shape implements IShape {
  protected x: number
  protected y: number
  protected color: string

  constructor(params: ShapeParams) {
    this.x = params.x
    this.y = params.y
    this.color = params.color
  }
  
  abstract clone(): any
  abstract square(): number
  abstract deepEquals(otherInstance: any): boolean
  
  getX(): number { return this.x }
  getY(): number { return this.y }
  getColor(): string { return this.color }
}